diffkemp/simpll/ModuleComparator.cpp

```diff
@@ -62,10 +76,16 @@ void ModuleComparator::compareFunctions(Function *FirstFun,
                                          showAsmDiffs, &GS, DI, this);
     if (fComp.compare() == 0) {
         DEBUG_WITH_TYPE(DEBUG_SIMPLL,
-                        dbgs() << "Function " << FirstFun->getName()
-                               << " is same in both modules\n");
+            decreaseDebugIndentLevel();
+            dbgs() << getDebugIndent() << "Functions are equal\n";
+        );
         ComparedFuns.at({FirstFun, SecondFun}) = Result::EQUAL;
     } else {
+        DEBUG_WITH_TYPE(DEBUG_SIMPLL,
+            decreaseDebugIndentLevel();
+            dbgs() << getDebugIndent() << "Functions are not equal\n";
+            increaseDebugIndentLevel()
```

Comment:
I don't think that calling `increaseDebugIndentLevel` here is correct. In case the result is "not equal" but there are no functions to inline, the indent level will never be decreased. I suggest moving this call into the following loop.

Reply:
Yes, I've missed this one. I believe this is also the reason @lenticularis39 ended up having an error (`getDebugIndent` in the while loop was executed with a negative number). Maybe I should protect the `debugIndentLevel` variable from being negative and/or make it unsigned so that this cannot result in a crash in the future?

Reply:
I agree, make it unsigned. However, its value should never underflow, therefore, add an assertion to `decreaseDebugIndentLevel`:

    assert(debugIndentLevel >= 0);

This will help us to find this kind of errors earlier.